cmake/define_libwiredtiger.cmake

```diff
@@ -0,0 +1,103 @@
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#  All rights reserved.
+#
+#  See the file LICENSE for redistribution information
+#
+
+# define_wiredtiger_library(target type)
+# A helper that defines a wiredtiger library target. This defining a set of common targets and properties we
+# want to associated to any given 'libwiredtiger' target. Having this as a macro allows us to de-duplicate common
+# definitions if want to create multiple versions of libwiredtiger i.e. static and shared builds. Note: this
+# macro assumes you only call it once per libwiredtiger flavour (e.g. only one 'static' definition), use carefully.
+#   target - Name of the libwiredtiger target.
+#   type - Library type of wiredtiger target e.g. STATIC, SHARED or MODULE.
+#   SOURCES - Sources to be compiled under the wiredtiger library. Requires that at least one source file/object is defined.
+#   PUBLIC_INCLUDES - Public interface includes of the wiredtiger library.
+#   PRIVATE_INCLUDES - Private interface includes of the wiredtiger library.
+macro(define_wiredtiger_library target type)
+    cmake_parse_arguments(
+        "DEFINE_WT"
+        ""
+        ""
+        "SOURCES;PUBLIC_INCLUDES;PRIVATE_INCLUDES"
+        ${ARGN}
+    )
+    if (NOT "${DEFINE_WT_UNPARSED_ARGUMENTS}" STREQUAL "")
+        message(FATAL_ERROR "Unknown arguments to define_wiredtiger_library: ${DEFINE_WT_UNPARSED_ARGUMENTS}")
+    endif()
+    if ("${DEFINE_WT_SOURCES}" STREQUAL "")
+        message(FATAL_ERROR "No sources given to define_wiredtiger_library")
+    endif()
+
+    # Define the wiredtiger library target.
+    add_library(${target} ${type} ${DEFINE_WT_SOURCES})
+    # Append any include directories to the library target.
+    if(DEFINE_WT_PUBLIC_INCLUDES)
+        target_include_directories(${target} PUBLIC ${DEFINE_WT_PUBLIC_INCLUDES})
+    endif()
+    if(DEFINE_WT_PRIVATE_INCLUDES)
+        target_include_directories(${target} PRIVATE ${DEFINE_WT_PRIVATE_INCLUDES})
+    endif()
+    # Append any provided C flags.
+    if(COMPILER_DIAGNOSTIC_C_FLAGS)
```

Comment:
We recently added `COMPILER_DIAGNOSTIC_CXX_FLAGS`, should we add them here too?

Reply:
Good question, I don't believe we need them here. The CXX flags should only be used for C++ targets. Given `libwiredtiger` is entirely a C target, we shouldn't need to append them.

Reply:
Sounds good!